src/Core/src/Platform/Android/LocalizedDigitsKeyListener.cs

```diff
@@ -1,32 +1,31 @@
-using System.Collections.Generic;
+﻿using System.Collections.Generic;
 using Android.Text;
 using Android.Text.Method;
 using Java.Lang;
 using Java.Text;
 
-namespace Microsoft.Maui.Controls.Compatibility.Platform.Android
+namespace Microsoft.Maui.Platform.Android
 {
-	internal class LocalizedDigitsKeyListener : NumberKeyListener
+	public class LocalizedDigitsKeyListener : NumberKeyListener
 	{
 		readonly char _decimalSeparator;
 
 		// I'm not aware of a situation/locale where this would need to be something different, 
 		// but we'll make it easy to localize the sign in the future just in case
 		const char SignCharacter = '-';
 
-		static Dictionary<char, LocalizedDigitsKeyListener> s_unsignedCache;
-		static Dictionary<char, LocalizedDigitsKeyListener> s_signedCache;
+		static Dictionary<char, LocalizedDigitsKeyListener>? s_unsignedCache;
+		static Dictionary<char, LocalizedDigitsKeyListener>? s_signedCache;
 
 		static char GetDecimalSeparator()
 		{
-			var format = NumberFormat.Instance as DecimalFormat;
-			if (format == null)
-			{
+			if (!(NumberFormat.Instance is DecimalFormat format))
 				return '.';
-			}
 
-			DecimalFormatSymbols sym = format.DecimalFormatSymbols;
-			return sym.DecimalSeparator;
+
+			DecimalFormatSymbols? sym = format.DecimalFormatSymbols;
+
+			return sym == null ? '.' : sym.DecimalSeparator;
 		}
 
 		public static NumberKeyListener Create(InputTypes inputTypes)
```

Comment:
```suggestion
using System.Collections.Generic;
﻿using System.Collections.Generic;
 using Android.Text;
 using Android.Text.Method;
 using Java.Lang;
 using Java.Text;
 
namespace Microsoft.Maui.Controls.Compatibility.Platform.Android
namespace Microsoft.Maui.Platform.Android
 {
	internal class LocalizedDigitsKeyListener : NumberKeyListener
	public class LocalizedDigitsKeyListener : NumberKeyListener
 	{
 		readonly char _decimalSeparator;
 
 		// I'm not aware of a situation/locale where this would need to be something different, 
 		// but we'll make it easy to localize the sign in the future just in case
 		const char SignCharacter = '-';
 
		static Dictionary<char, LocalizedDigitsKeyListener> s_unsignedCache;
		static Dictionary<char, LocalizedDigitsKeyListener> s_signedCache;
		static Dictionary<char, LocalizedDigitsKeyListener>? s_unsignedCache;
		static Dictionary<char, LocalizedDigitsKeyListener>? s_signedCache;
 
 		static char GetDecimalSeparator()
 		{
			var format = NumberFormat.Instance as DecimalFormat;
			if (format == null)
			{
			if (!(NumberFormat.Instance is DecimalFormat format))
 				return '.';
			}
 
			DecimalFormatSymbols sym = format.DecimalFormatSymbols;
			return sym.DecimalSeparator;

			DecimalFormatSymbols? sym = format.DecimalFormatSymbols;

			return sym == null ? '.' : sym.DecimalSeparator;
 		}
 
 		public static NumberKeyListener Create(InputTypes inputTypes)
```